packages/eslint/src/generators/convert-to-flat-config/generator.spec.ts

```diff
@@ -430,4 +431,63 @@ describe('convert-to-flat-config generator', () => {
       "
     `);
   });
+
+  it('should convert project if target is defined via plugin as string', async () => {
+    await lintProjectGenerator(tree, {
+      skipFormat: false,
+      linter: Linter.EsLint,
+      project: 'test-lib',
+      setParserOptionsProject: false,
+    });
+    updateJson(tree, 'nx.json', (json: NxJsonConfiguration) => {
+      delete json.targetDefaults;
+      json.plugins = ['@nx/eslint/plugin'];
+      return json;
+    });
+    updateJson(
+      tree,
+      'libs/test-lib/project.json',
+      (json: ProjectConfiguration) => {
+        delete json.targets.lint;
+        return json;
+      }
+    );
+
+    await convertToFlatConfigGenerator(tree, options);
```

Comment:
Please could we add an assertion just before that the flat config doesn't already exist? Verifying that mentally requires taking in the whole rest of the file right now